src/qt/optionsmodel.cpp

```diff
@@ -64,6 +66,15 @@ void OptionsModel::Init(bool resetSettings)
     if (!settings.contains("fMinimizeToTray"))
         settings.setValue("fMinimizeToTray", false);
     fMinimizeToTray = settings.value("fMinimizeToTray").toBool() && !fHideTrayIcon;
+#else
+    // On systems without tray icon functionality we set some default values
+    // and remove settings from previous versions of the software to not
+    // confuse people looking into the .config or .plist file.
+    fHideTrayIcon = true;
+    settings.remove("fHideTrayIcon");
```

Comment:
IIRC we don't remove old/unused settings. ping @laanwj.

Reply:
Ok, I have removed the removal of unused settings. To be honest, I realized the "ping @laanwj" only after my commit. Let's hope, Wladimir agrees.

Reply:
Yes I agree, this makes it possible to upgrade/downgrade without losing settings.